cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -677,6 +680,12 @@ def _validate_dataframe(self, df_name: str):
                         self.warnings.append(column_def["warning_message"])
                     self._validate_column(column, column_name, df_name, column_def)
 
+        # Validate user-provided column names
+        for column in df.columns:
+            if re.match(r"__", column):  # re.match matches strictly from beginning of str
+                self.errors.append(f"The column '{column}' in dataframe '{df_name}' is not valid; columns must not "
+                                   f"have a '__' prefix.")
+
```

Comment:
right now this means `var`, `raw.var`, and `obs`, since they are the 3 components of type `dataframe` in the schema 

see [conversation](https://czi-sci.slack.com/archives/C03PXFMM7ST/p1695255977203039?thread_ts=1694818587.348629&cid=C03PXFMM7ST)